servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererObserver.java

```diff
@@ -46,6 +60,18 @@ interface DnsDiscoveryObserver {
          * @return {@link DnsResolutionObserver} that provides visibility into results of the current DNS resolution
          */
         DnsResolutionObserver onNewResolution(String name);
+
+        /**
+         * Notifies that the current DNS discovery got canceled (did not complete successfully).
+         */
+        default void discoveryCanceled() { } // FIXME: 0.43 remove default
```

Comment:
I know that both versions are grammatically correct. Only for consistency with other public API we have, consider using double LL: `discoveryCancelled()`.

Example:
https://github.com/apple/servicetalk/blob/d45724ddbda177b91d0426b7737757e506218bfb/servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/SequentialCancellable.java#L84-L85